### ISSUE TITLE
Improve global search (note)

### ISSUE THREAD

**Issue by seansan** [OpenMage/magento-lts#134]
This is a note to improve global search. If someone already has some working code then this is great. 

Things to improve, maybe there is more

- replace search with %STRING%, orders numbers like %STRING (often we only get last couple of digits xxx not the leading 300000xxxx)
- search for lastname, firstname, email, order number
- if ordernumber found then show orders first
- if customer data used to localize then show customer data first
- improve 2nd line small font span element (giving extra information)
- add store name to order number in results
- sort by last created orders first (this is probably what the customer is calling about)
- add some index because the search can be really really slow (try using it with someone on the telephone)

**Comment by colinmollenhour** [comment]
We had to improve global search by completely disabling it.. On a large store it becomes completely unusable.. Maybe with some FULLTEXT indexes it could be made usable but in general using `LIKE %..` only works for smaller stores where full table scans are doable. On another installation it wasn't so bad when we replaced all of the `LIKE %...%` with `LIKE ...%` so that indexes are used.

**Comment by seansan** [comment]
I agree - haha turning it off always makes it sound bad. There must be a working solution. Did you ever find alternatives? Or maybe we should add LIKE ...% & maybe even add an index here and there

My post was mainly to capture: 
- what do we find important to search for
- how do we speed this up (query + add. index)
- how can we better present it

**Comment by colinmollenhour** [comment]
We have do not have an alternative for global search, we just quit using it in favor of improving grid search. I changed the default grid searching behavior to use `LIKE ...%` by default (to be faster) and added some magic to it so that it supports things like IN, REGEX, LIKE %...% (by specifying *...* for the lay-person), etc.

**Comment by colinmollenhour** [comment]
I think ideally it would be replaced with something like elasticsearch but I think that it out of the scope of LTS project.

**Comment by seansan** [comment]
@colinmollenhour I read that you did some manual changes to your installation. Is there something to move/push to LTS ... or should we rather close the issue?

(for now we dont use it either)

**Comment by sreichel** [comment]
Any updates here? Make it configurable?

**Comment by colinmollenhour** [comment]
I think just replacing all `LIKE %...%` with `LIKE ...%` makes sense because it is no good if it isn't performant. All of the other suggestions @seansan had generally sound good so long as they don't impact performance. In general the rule should be that global search only does queries that are able to effectively use an index.

**Comment by seansan** [comment]
Not sure what to do here. We completely rewrote search based on above and specificlly for our site (and products). Great improvements can be made here though .... 

- Search now finds customers by lastname or last digit of the order # (and shows the last order)
- Search now finds products by SKU and EAN/Barcode + shows stock
- It is fast

I am unable to add this objectively and more specifically in a well coded version. I can however share what we have, think it out if someone feels up to it

**Comment by ADDISON74** [comment]
OFF TOPIC: On the frontend side the search leaves much to be desired. I tested all the settings that Magento offers and I didn't improve much. I found an extension that radically improves search results. It's called Active CatalogSearch 1.1.0. Maybe in the future I will move to a solution based on ElasticSearch.
